content/guide/extending-classes-and-conforming-to-protocols-ios.md

```diff
@@ -0,0 +1,229 @@
+    // We will obtain the method signature from the protocol.
+    applicationDidFinishLaunchingWithOptions: function (application, launchOptions) {
+        this._window = new UIWindow(UIScreen.mainScreen.bounds);
+        this._window.rootViewController = MyViewController.alloc().init();
+        this._window.makeKeyAndVisible();
+        return true;
+    }
+}, {
+    // The name for the registered Objective-C class.
+    name: "MyAppDelegate",
+    // Declare that the native Objective-C class will implement the UIApplicationDelegate Objective-C protocol.
+    protocols: [UIApplicationDelegate]
+});
+```
+
+Let's look how to declare a delegate in Typescript by setting one for the [Tesseract-OCR-iOS](https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS/6510b29bbf18655f29a26f484b00a24cc66ed88b) API
+
+```ts
+interface G8TesseractDelegate extends NSObjectProtocol {
+    preprocessedImageForTesseractSourceImage?(tesseract: G8Tesseract, sourceImage: UIImage): UIImage;
+    progressImageRecognitionForTesseract?(tesseract: G8Tesseract): void;
+    shouldCancelImageRecognitionForTesseract?(tesseract: G8Tesseract): boolean;
+}
+```
+
+Implementing the delegate:
+
+```ts
+@NativeClass()
+class G8TesseractDelegateImpl
+    extends NSObject // native delegates mostly always extend NSObject
+    implements G8TesseractDelegate {
+
+    static ObjCProtocols = [G8TesseractDelegate] // define our native protocols
+
+    static new(): G8TesseractDelegateImpl {
+        return <G8TesseractDelegateImpl>super.new() // calls new() on the NSObject
+    }
+
+    preprocessedImageForTesseractSourceImage(tesseract: G8Tesseract, sourceImage: UIImage): UIImage {
+        console.info('preprocessedImageForTesseractSourceImage')
+        return sourceImage
+    }
+
+    progressImageRecognitionForTesseract(tesseract: G8Tesseract) {
+        console.info('progressImageRecognitionForTesseract')
+    }
+
+    shouldCancelImageRecognitionForTesseract(tesseract: G8Tesseract): boolean {
+        console.info('shouldCancelImageRecognitionForTesseract')
+        return false
+    }
+
+}
+```
+
+Using the class conforming to the `G8TesseractDelegate`:
+
+```ts
+function image2text(image: UIImage): string {
+    let delegate: G8TesseractDelegateImpl = G8TesseractDelegateImpl.new()
+    let tess: G8Tesseract = G8Tesseract.new()
+    tess.delegate = delegate
+    /*=============================
+    =            NOTES            =
+    =============================*/
+    // The `tess.delegate` property is weak and won't be retained by the Objective-C runtime so you should manually keep the delegate JS object alive as long the tessaract instance is alive
+    /*=====  End of NOTES  ======*/
+    tess.image = image
+    let results: boolean = tess.recognize()
+    if (results == true) {
+        return tess.recognizedText
+    } else {
+        return 'ERROR'
+    }
+}
+
+```
+## Limitations
+- You shouldn't extend an already extended class
```

Comment:
It it better to use "should not" instead of "shouldn't"?  The same applies with the use of "can't".